packages/rocketchat-language-plugin/server/add_or_update_user.js

```diff
@@ -0,0 +1,26 @@
+Meteor.methods({
+	add_or_update_user(channel, plugin_name) {
+		const channel_exist = Meteor.users.findOne({
+			$and: [{ _id: Meteor.userId()}, {'automatic_channels.name': channel}]
+		});
+		if (!channel_exist) {
+			Meteor.users.update(Meteor.userId(), {
+				$push: {
+					automatic_channels: {'name': channel, 'plugin': plugin_name, 'added_to':'true', 'blacklisted': 'false'}
+				}
+			});
+		} else {
+			Meteor.users.update({
+				_id: Meteor.userId(),
+				'automatic_channels.name': channel
+			}, {
+				$set: {
+					'automatic_channels.$.added_to' :'true'
```

Comment:
added_to is true when user is added to an automatic channel, but once the user is removed from that channel because of being added to another channel( when he might have changed his language preference) 'added_to' changed to false